lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -349,12 +357,93 @@ def get_parent_of(record, path)
     [parent_object, path[-1]]
   end
 
-  def send_bulk(data)
+  def send_bulk(data, count)
     retries = 0
     begin
       response = client.bulk body: data
       if response['errors']
-        log.error "Could not push log to Elasticsearch: #{response}"
+        errors = Hash.new(0)
+        errors_bad_resp = 0
+        errors_unrecognized = 0
+        successes = 0
+        duplicates = 0
+        bad_arguments = 0
+        # Count up the individual error types returned for each item
+        # Note: this appears to be an undocumented feature of Elasticsearch
+        # https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-bulk.html
+        # When you submit an "index" write_operation, with no "_id" field in the
+        # metadata header, Elasticsearch will turn this into a "create"
+        # operation in the response.
+        response['items'].each do |item|
+          if item.has_key?(@write_operation)
```

Comment:
@jcantrill can you be more specific?  Most of these `if/elsif/else` cases don't involve a single target.  The only one that I can see that would be applicable is the one beginning on line https://github.com/ViaQ/fluent-plugin-elasticsearch/pull/2/files#diff-ae62a005780cc730c558e3e4f47cc544R396
But even then I'm not sure how to write the first case
```
case status
when (409 && CREATE_OP == write_operation) then ...
```
A simple test like this fails:
```
irb
irb(main):001:0> ii = 0
=> 0
irb(main):002:0> foo = "foo"
=> "foo"
irb(main):003:0> case ii
irb(main):004:1> when (foo == "foo" && 0) then puts "here"
irb(main):005:1> end
here
=> nil
```
I'm not familiar with `case/when` so if I'm missing something, please let me know.